app/queue.py

```diff
@@ -140,6 +154,33 @@ def __register_scripts(self):
             """
         )
 
+        self.scripts[self.LUA_MOVE_FROM_INFLIGHT] = self._redis_client.register_script(
+            """
+            local DEFAULT_CHUNK = 99
+
+            local source        = ARGV[1]
+            local destination   = ARGV[2]
+            local count         = tonumber(redis.call("LLEN", source))
+
+            local chunk_size    = math.min(math.max(0, count-1), DEFAULT_CHUNK)
+            local current       = 0
+            local all           = {}
+
+            while current < count do
+                local elements = redis.call("LRANGE", source, 0, chunk_size)
+                redis.call("LPUSH", destination, unpack(elements))
+                redis.call("LTRIM", source, chunk_size+1, -1)
+                for i=1,#elements do all[#all+1] = elements[i] end
+
+                current    = current + chunk_size+1
+                chunk_size = math.min((count-1) - current, DEFAULT_CHUNK)
+            end
```

Comment:
Normally, after executing the script, there should be no more inflight list. Hence deleting it afterward in Python land should not be necessary.